batch/batch/spec_writer.py

```diff
@@ -0,0 +1,56 @@
+import secrets
+import logging
+
+log = logging.getLogger('batch.spec_writer')
+
+
+class SpecWriter:
+    byteorder = 'little'
+    signed = False
+    bytes_per_offset = 8
+
+    @staticmethod
+    def get_index_file_offsets(job_id, start_job_id):
+        assert job_id >= start_job_id
+        bytes_per_job = 2 * SpecWriter.bytes_per_offset
```

Comment:
I think you can get away with one offset per job, right?  Record the start, and then the end is the is the start of the next job (or None, which means you read to the end of the file for the last one.  Alternatively, you can store n+1 offsets, with the last offset being the size of the file.)